source/js/main.js

```diff
@@ -444,6 +444,11 @@ const createRecipeClicked = () => {
   if (currSavedPageSelect === 'List 2') {
     // add card to saved recipe page
     const grid = document.querySelector('.saved-recipes .results-grid');
+    const possibleRepeatCard = grid.querySelectorAll(`.id_${finalObject.id}`);
```

Comment:
It might be helpful to describe what this line is doing, since the bug description in the linked issue isn't comprehensive.

Reply:
to sync saved recipe page when user create a recipe, we need to create a card in saved recipe page when creating a recipe. local storage doesn't allow duplicates so if you create two same recipe, storage will keep the latest version. But we didn't check for duplicate in current saved recipe page. If you don't refresh the page right now and create two same recipes. although only the later is saved in storage, you will still see duplicate card under saved recipe page. this line checks and remove display repeat in saved recipe page